packages/ra-core/src/form/FormWithRedirect.tsx

```diff
@@ -92,7 +93,38 @@ const FormWithRedirect: FC<FormWithRedirectProps> = ({
         [save]
     );
 
-    const formContextValue = useMemo(() => ({ setOnSave }), [setOnSave]);
+    const formContextValue = useMemo<FormContextValue>(
+        () => ({
+            setOnSave,
+            getGroupFields: name => formGroups.current[name] || [],
+            registerGroup: name => {
+                formGroups.current[name] = formGroups.current[name] || [];
+            },
+            unregisterGroup: name => {
+                delete formGroups[name];
+            },
+            registerField: (source, group) => {
+                if (group) {
+                    formGroups.current[group] = formGroups.current[group] || [];
```

Comment:
why do you allow an input to register to a group that hasn't been registered here, while you forbid it in the following function?

Reply:
I found out that inputs were registering themselves before the group